File: FitToFit/FitToFit.Model/Treninzi.cs
namespace FitToFit.Model
{
    public partial class Treninzi
    {
        public int TreningId { get; set; }

        public string Naziv { get; set; } = null!;

        public string Opis { get; set; } = null!;

        public int MaxBrojClanova { get; set; }

        public decimal CijenaPoTerminu { get; set; }

        public string Trajanje { get; set; } = null!;

        public decimal ProsjecnaPotrosnjaKalorija { get; set; }

        public int VrstaId { get; set; }

        public string Namjena { get; set; } = null!;

        public byte[]? Slika { get; set; }
        public virtual ICollection<Termini> Terminis { get; set; } = new List<Termini>();

        public virtual ICollection<TreninziVjezbe> TreninziVjezbes { get; set; } = new List<TreninziVjezbe>();
    }
}

File: FitToFit/UI/fittofit_admin/lib/models/treninzi.dart
import 'package:fittofit_admin/models/termini.dart';
import 'package:fittofit_admin/models/vjezbeTreninzi.dart';
import 'package:json_annotation/json_annotation.dart';

part 'treninzi.g.dart';

@JsonSerializable()
class Treninzi {
  int treningId;
  String naziv;
  String opis;
  int maxBrojClanova;
  double cijenaPoTerminu;
  String trajanje;
  double prosjecnaPotrosnjaKalorija;
  int vrstaId;
  String namjena;
  String? slika;
  List<Termini> terminis = [];
  List<VjezbeTreninzi>? treninziVjezbes = [];

  Treninzi(
      {required this.treningId,
      required this.naziv,
      required this.opis,
      required this.maxBrojClanova,
      required this.cijenaPoTerminu,
      required this.trajanje,
      required this.prosjecnaPotrosnjaKalorija,
      required this.vrstaId,
      required this.namjena,
      this.slika,
      required this.terminis,
      this.treninziVjezbes});

  factory Treninzi.fromJson(Map<String, dynamic> json) =>
      _$TreninziFromJson(json);

  Map<String, dynamic> toJson() => _$TreninziToJson(this);
}
